pkg/policy/selectorcache.go

```diff
@@ -258,6 +267,27 @@ func (s *selectorManager) IsWildcard() bool {
 	return s.key == wildcardSelectorKey
 }
 
+// must be holding selectorcache mutex
+func (s *fqdnSelector) NotifyAdded() {
+	// Make the user (FQDN subsystem) aware of this selector.
+	if s.user == nil {
+		return
+	}
+
+	if ids, exists := s.user.StartManagerFQDNSelector(s.selector); !exists {
+		for _, id := range ids {
```

Comment:
Why do we set the cached selections only if `!exists`? Maybe `ids` is empty or `nil` if `exists`, but even then it would be simpler to just iterate over `ids` unconditionally. That way there would be no question about potentially missing some selections.

Reply:
Good point, will do that.